spark/src/main/scala/org/apache/spark/shuffle/sort/RowPartition.scala

```diff
@@ -30,7 +30,11 @@ class RowPartition(initialSize: Int) {
     rowSizes += size
   }
 
-  def getNumRows: Int = rowAddresses.size
+  def getNumRows: Int = if (rowAddresses == null) {
+    0
+  } else {
+    rowAddresses.size
+  }
```

Comment:
This is unrelated. Just when I tested several memory configs to try to trigger spilling, this method throws NPE sometimes.